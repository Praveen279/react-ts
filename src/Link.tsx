import React, { useEffect, useState } from 'react';

interface LinkProps {
  className?: string;
}

const Link: React.FC<LinkProps> = ({
  className
}) => {
  const [counter, setCounter] = useState(0);
  useEffect(() => {
    if (counter == 0) {
      setCounter(counter + 1);
    }
  });
  return (
    <div
      className={className}
      onClick={() => setCounter(counter + 1)}
    >
      {counter}
    </div>
  );
}

export default Link;