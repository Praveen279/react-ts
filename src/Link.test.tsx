import React from "react";
import { mount } from 'enzyme';
import Link from './Link';

describe('Link', () => {
  it('Test useEffect', () => {
    const link = mount(<Link className="my-link-class">Link to Google</Link>);
    expect(link).toMatchSnapshot();
    link.find('div').simulate('click');
    expect(link).toMatchSnapshot();
  });
});