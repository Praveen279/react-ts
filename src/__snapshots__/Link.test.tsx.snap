// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Test useEffect 1`] = `ReactWrapper {}`;

exports[`Link Test useEffect 2`] = `ReactWrapper {}`;
